crypto/src/swap.rs

```diff
@@ -1,13 +1,43 @@
 use crate::ka;
 use crate::transaction::Fee;
 use anyhow::Result;
+use chacha20poly1305::{
+    aead::{Aead, NewAead},
+    ChaCha20Poly1305, Key, Nonce,
+};
+use decaf377::FieldExt;
+use once_cell::sync::Lazy;
 use penumbra_proto::{dex as pb, Protobuf};
 
 use crate::asset::Id as AssetId;
 use crate::keys::OutgoingViewingKey;
 
-// TODO: not sure what byte size is necessary here yet
-pub const SWAP_CIPHERTEXT_BYTES: usize = 128;
+// Swap ciphertext byte length
+pub const SWAP_CIPHERTEXT_BYTES: usize = 169;
+// Swap plaintext byte length
+pub const SWAP_LEN_BYTES: usize = 153;
+pub const OVK_WRAPPED_LEN_BYTES: usize = 80;
+
+/// The nonce used for swap encryption.
+///
+/// The nonce will always be `[0u8; 12]` which is okay since we use a new
+/// ephemeral key each time.
+pub static SWAP_ENCRYPTION_NONCE: Lazy<[u8; 12]> = Lazy::new(|| [0u8; 12]);
+
+// Can add to this/make this an enum when we add additional types of swaps.
+// TODO: is this actually something we would do? suppose it doesn't hurt to build this
+// in early.
+pub const SWAP_TYPE: u8 = 0;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("Swap type unsupported")]
+    SwapTypeUnsupported,
+    #[error("Swap deserialization error")]
+    SwapDeserializationError,
+    #[error("Decryption error")]
+    DecryptionError,
+}
```

Comment:
If we don't need to use these for control flow, we could drop the enum and just use `anyhow` to make ad-hoc errors.